### PR TITLE
Refactored to follow Origami spec

### PR DESCRIPTION
**Pull request by AlbertoElias**
cc @alicebartlett @keirog 

So this will hopefully be step 2 in the refactor. I've been working on making the header follow the Origami spec and have only modified things on that basis.

Next step will be to look into simplifying DOM and Sass a bit more, and add silent mode. And then I'll add the minimal header theme.

Concerns with this: 

* Having to use `o-header__meganav-section-head` class more than I would have liked. Probably not worth looking more into it for this version so Markets Data can get their hands on it asap and it can be looked into in the DOM and Sass simplification
* Having tested in IE or mobile browsers, works fine in Chrome, Safari and Firefox desktop

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,52 +1,40 @@
...
-make build
-```
-
-This will also rebuild the navigation templates using data from `next-navigation`.
-
-## Feature flags
-
-To use feature flags pass in an instance of the next-feature-flags-client (which should already have been initialised) like so:
+* editionswitchClassName:
+* headerClassName:
```
Comment (triblondon) [inline README.md]: More needed here?

File: README.md [hunk 2]
```diff
@@ -1,52 +1,40 @@
...
+* data-o-header-search
+* data-o-header--js
+* data-o-header-form-search
+    - input
+    - label
+* data-o-header-togglable-search
+* data-o-header-edition-switch-button
+* data-o-header-selectable
+* data-o-header-togglable
+* data-o-header-togglable-nav
```
Comment (triblondon) [inline README.md]: Do we need to document the classes here as well?

File: demos/src/main.mustache [hunk 3]
```diff
@@ -1,477 +1,489 @@
...
-						<ul class="edition-switcher__editions" aria-hidden>
-							<li class="edition-switcher__edition">
-									<a href="/international?edition=international" class="edition-switcher__edition__link" data-trackable="international">International Edition</a>
-							</li>
-						</ul>
-					</div>
-				<div class="o-header__masthead__logo">
-					<a class="o-header__logo__link" data-trackable="logo-large" href="/" title="Go to Financial Times homepage">
-						<span class="n-util-visually-hidden">
+<header class="o-header o-header--light-on-dark" data-o-component="o-header">
```
Comment (triblondon) [inline demos/src/main.mustache]: Future suggestion: store data separately and template it into the main.mustache to reduce repetition in the template.

File: demos/src/main.scss [hunk 4]
```diff
@@ -1 +1,13 @@
...
+
+body {
+	margin: 0;
+	font-family: "MetricWeb";
+}
+
+.core .o--if-js,
+.enhanced .o--if-no-js {
+	display: none !important;
+}
```
Comment (triblondon) [inline demos/src/main.scss]: Isn't this added by obt demo as part of the standard demo wrapper template?

File: main.js [hunk 5]
```diff
@@ -1,17 +1,10 @@
...
+	oHeader.init();
+	document.removeEventListener('o.DOMContentLoaded', constructAll);
 };
 
-export default {
-	init
-};
+document.addEventListener('o.DOMContentLoaded', constructAll);
+
+export default oHeader;
```
Comment (triblondon) [inline main.js]: Whoa.  ES6 modules?  Not convinced by this - aren't we sticking with commonJS?

**Comment by triblondon** [comment]
OK, this looks overall like fantastic work.  I can't do a detailed review due to the size, and I think there are a few issues here that can be deferred to a future release.  The basic architecture looks sound.

File: main.js (hunk 5 above)
Comment (AlbertoElias) [inline main.js]: We're already using it in some modules for tests, and Next is heavily using it. The spec is stabilising as well and it's going to be implemented in browsers

File: demos/src/main.scss (hunk 4 above)
Comment (AlbertoElias) [inline demos/src/main.scss]: Oh, true, it adds `body { margin: 0; } .demo-js .o--if-nojs { display: none; }`; Should I change .demo-js to enhanced?

File: demos/src/main.mustache (hunk 3 above)
Comment (AlbertoElias) [inline demos/src/main.mustache]: Yup, I had that in mind

File: README.md (hunk 2 above)
Comment (AlbertoElias) [inline README.md]: I haven't really worked on the documentation. I just added the stuff that I knew I would have to elaborate on so I wouldn't forget. I don't think I'll add the classes though, there are too many. For modules like this, we just suggest in the readme that people can check out the demo markup

File: README.md (hunk 1 above)
Comment (AlbertoElias) [inline README.md]: Like I mentioned below, haven't worked on it yet

**Pull request merged**